pipeline/data_store.py

```diff
@@ -429,7 +495,7 @@ def get_provenance(self, process, pars_dict, upstream_provs=None, session=None):
 
```

Comment:
Expanding the code up a bit from this diff -- it looks like it tries (a) to find a provenance in `upstream_provs` (which is initialized from `self.upstream_provs`, (b) to pull a provenance from a object produced by the right step, and (c) if that fails, to just the "latest" one from the database.

Before that last step, shouldn't it do a recursive call to DataStore.get_provenance, passing the upstream step as the process and the appropriate thing from the config file as `pars_dict`?  I'd have to think harder about this, but upon first glance it seems to me that the configured upstream parameters ought to somehow come into the fallback provenance generation.

Reply:
(In practice, this will not matter, at least when running the full pipeline, because of ` make_provenance_tree` in `top_level.py`.)

Reply:
yes that would just be pretty complicated as the datastore doesn't have access to all the pipeline objects/configuration to do that sort of thing backwards. 
I don't particularly like this fallback but I couldn't find something less complicated to make it work. We could just make a place to put the whole provenance tree before you start, but that would make it hard to just get back a datastore from a single pipleine object, i.e., `ds = preprocessor(exposure, section_id)` would not work that way.